probe-rs/src/architecture/arm/communication_interface.rs

```diff
@@ -77,19 +78,19 @@ pub trait ArmProbeInterface: DapAccess + SwdSequence + SwoAccess + Send {
     /// Returns a memory interface to access the target's memory.
     fn memory_interface(
         &mut self,
-        access_port: MemoryAp,
+        access_port: &MemoryAp,
     ) -> Result<Box<dyn ArmProbe + '_>, ArmError>;
 
     /// Returns information about a specific access port.
-    fn ap_information(&mut self, access_port: GenericAp) -> Result<&ApInformation, ArmError>;
+    fn ap_information(&mut self, access_port: &GenericAp) -> Result<&ApInformation, ArmError>;
 
     /// Return the currently connected debug port.
     fn current_debug_port(&self) -> DpAddress;
 
     /// Returns the number of access ports the debug port has.
     ///
     /// If the target device has multiple debug ports, this will switch the active debug port
-    /// if necessary. This will also  
+    /// if necessary. This will also
```

Comment:
I would actually leave the two spaces here just to make it clear (if trailing spaces are highlighted in your editor) that something's missing from this sentence.

Reply:
Oh, I don’t know why this got removed (the spaces). I removed this change from the PR.

For what it’s worth, this method gets removed in a future PR, so the incomplete comment isn’t much of an issue.